File: package/main/src/String/deburr.ts
/**
 * Removes diacritical marks from a string by decomposing characters and
 * stripping the combining marks (e.g. "é" becomes "e").
 * @param {string} string_ - Input string
 * @returns {string} String without diacritical marks
 * @example
 * deburr("déjà vu"); // "deja vu"
 * deburr("Crème brûlée"); // "Creme brulee"
 */
export const deburr = (string_: string): string =>
  string_.normalize("NFD").replaceAll(/\p{Diacritic}/gu, "");

File: package/umt_python/src/string/deburr.py
import unicodedata


def deburr(string_: str) -> str:
    """
    Removes diacritical marks from a string by decomposing characters and
    stripping the combining marks (e.g. "é" becomes "e").

    Args:
        string_: Input string.

    Returns:
        String without diacritical marks.

    Example:
        >>> deburr("déjà vu")
        'deja vu'
        >>> deburr("Crème brûlée")
        'Creme brulee'
        >>> deburr("résumé café")
        'resume cafe'
        >>> deburr("hello world")
        'hello world'
    """
    decomposed = unicodedata.normalize("NFD", string_)
    return "".join(char for char in decomposed if not unicodedata.combining(char))
